src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -222,6 +223,45 @@ private Map<String, FormDataBodyPart> extractFilesFromMultiPart(FormDataMultiPar
     return files;
   }
 
+  /**
+   * This endpoint updates the dataset.
+   */
+  @PUT
+  @Consumes({MediaType.MULTIPART_FORM_DATA})
+  @Produces({MediaType.APPLICATION_JSON})
+  @Path("/v3/{datasetId}")
+  @RolesAllowed({ADMIN, CHAIRPERSON})
+  public Response updateByDatasetUpdate(
+      @Auth AuthUser authUser,
+      @PathParam("datasetId") Integer datasetId,
+      FormDataMultiPart multipart,
+      @FormDataParam("dataset") String json) {
+
+    try {
+
+      DatasetUpdate update = new Gson().fromJson(json, DatasetUpdate.class);
```

Comment:
I think you may need to try-catch in case the JSON in invalid, too.

Reply:
I created a test for this case using the current try-catch, and the response was a 400 which I would expect. Do you think it's redundant to add another try-catch? or am I looking for a different response?

Reply:
Oh yeah I missed that, that works.